File: backend/schemas/health.py
"""
Health schemas - Request/Response models for health tracking
"""
from pydantic import BaseModel
from uuid import UUID
from datetime import date, datetime
from typing import Optional


class CreateHealthEntryRequest(BaseModel):
    """Create health entry request"""
    date: date
    water: int = 0
    steps: int = 0
    calories: Optional[int] = None
    meals: list[str] = []
    meal_types: list[str] = []
    cycle: Optional[str] = None
    period_day: Optional[int] = None
    bath: bool = False
    face_wash: bool = False
    notes: Optional[str] = None
    
    class Config:
        json_schema_extra = {
            "example": {
                "date": "2024-01-15",
                "water": 6,
                "steps": 8420,
                "calories": 1800,
                "meals": ["Morning grains", "Noon feast"],
                "meal_types": ["healthy", "healthy"],
                "cycle": "waxing",
                "period_day": None,
                "bath": True,
                "face_wash": True,
                "notes": "Felt energetic"
            }
        }


class UpdateHealthEntryRequest(BaseModel):
    """Update health entry request"""
    water: Optional[int] = None
    steps: Optional[int] = None
    calories: Optional[int] = None
    meals: Optional[list[str]] = None
    meal_types: Optional[list[str]] = None
    cycle: Optional[str] = None
    period_day: Optional[int] = None
    bath: Optional[bool] = None
    face_wash: Optional[bool] = None
    notes: Optional[str] = None


class HealthEntryResponse(BaseModel):
    """Health entry response"""
    id: UUID
    user_id: UUID
    date: date
    water: int
    steps: int
    calories: Optional[int]
    meals: list[str]
    meal_types: list[str]
    cycle: Optional[str]
    period_day: Optional[int]
    bath: bool
    face_wash: bool
    notes: Optional[str]
    created_at: datetime
    updated_at: datetime


class HealthStatsResponse(BaseModel):
    """Health statistics response"""
    period: str  # weekly, monthly
    avg_water: float
    avg_steps: float
    avg_calories: Optional[float]
    total_days: int
    streak: int  # consecutive days with entries

File: src/api/health.ts
/**
 * Health API Service
 */
import { apiClient, PaginatedResponse } from './client';

export interface HealthEntry {
  id: string;
  user_id: string;
  date: string;
  water: number;  // glasses of water
  steps: number;
  calories?: number;
  meals: string[];
  meal_types: string[];
  cycle?: string;  // Moon phase: new, waxing, full, waning
  period_day?: number;
  bath: boolean;
  face_wash: boolean;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface CreateHealthEntryRequest {
  date: string;
  water?: number;
  steps?: number;
  calories?: number;
  meals?: string[];
  meal_types?: string[];
  cycle?: string;
  period_day?: number;
  bath?: boolean;
  face_wash?: boolean;
  notes?: string;
}

export interface UpdateHealthEntryRequest {
  water?: number;
  steps?: number;
  calories?: number;
  meals?: string[];
  meal_types?: string[];
  cycle?: string;
  period_day?: number;
  bath?: boolean;
  face_wash?: boolean;
  notes?: string;
}

export interface HealthStats {
  period: string;
  avg_water: number;
  avg_steps: number;
  avg_calories?: number;
  total_days: number;
  streak: number;
}

export const healthApi = {
  /**
   * Get health entries with pagination
   */
  getEntries: async (params?: {
    date?: string;
    page?: number;
    limit?: number;
  }): Promise<PaginatedResponse<HealthEntry>> => {
    const response = await apiClient.get<PaginatedResponse<HealthEntry>>('/health/entries', {
      params,
    });
    return response.data;
  },

  /**
   * Get health entry by ID
   */
  getEntry: async (id: string): Promise<HealthEntry> => {
    const response = await apiClient.get<HealthEntry>(`/health/entries/${id}`);
    return response.data;
  },

  /**
   * Create health entry
   */
  createEntry: async (data: CreateHealthEntryRequest): Promise<HealthEntry> => {
    const response = await apiClient.post<HealthEntry>('/health/entries', data);
    return response.data;
  },

  /**
   * Update health entry
   */
  updateEntry: async (id: string, data: UpdateHealthEntryRequest): Promise<HealthEntry> => {
    const response = await apiClient.put<HealthEntry>(`/health/entries/${id}`, data);
    return response.data;
  },

  /**
   * Delete health entry
   */
  deleteEntry: async (id: string): Promise<void> => {
    await apiClient.delete(`/health/entries/${id}`);
  },

  /**
   * Get health statistics
   */
  getStats: async (params: { start_date: string; end_date: string }): Promise<HealthStats> => {
    const response = await apiClient.get<HealthStats>('/health/stats', { params });
    return response.data;
  },
};
